src/contracts/core/DelegationManager.sol

```diff
@@ -206,20 +207,33 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         
         if (!stakerCanUndelegate(staker)) {
             // force the staker into "undelegation limbo" in the EigenPodManager if necessary
-            eigenPodManager.forceIntoUndelegationLimbo(staker);
+            uint256 podShares = eigenPodManager.forceIntoUndelegationLimbo(staker, operator);
+
+            IStrategy[] memory strategies;
+            uint256[] memory strategyShares;
 
             // force a withdrawal of all of the staker's shares from the StrategyManager
-            queuedWithdrawal = strategyManager.forceTotalWithdrawal(staker);
+            (strategies, strategyShares, queuedWithdrawal)
+                = strategyManager.forceTotalWithdrawal(staker);
+
+            _decreaseOperatorShares(operator, beaconChainETHStrategy, podShares);
+            for (uint i = 0; i < strategies.length; ) {
+                _decreaseOperatorShares(operator, strategies[i], strategyShares[i]);
 
-            // emit an event if this action was not initiated by the staker themselves
-            if (msg.sender != staker) {
-                emit StakerForceUndelegated(staker, operator);
+                unchecked {
+                    ++i;
+                }
             }
         }
 
         // actually undelegate the staker
         _undelegate(staker);
```

Comment:
Looks like we are only calling `_undelegate` here. no need to make it internal then

Reply:
good suggestion, IMO. addressed in commit https://github.com/Layr-Labs/eigenlayer-contracts/pull/183/commits/bf1de8a06817eb639c1c5a265b1aedfe0c9e1a33